File: backend/app/schemas/offer_letter.py
from datetime import datetime, date
from typing import Optional, List

from pydantic import BaseModel, Field

from app.models.offer_letter import OfferLetterStatus


class OfferLetterCreate(BaseModel):
    interview_id: int
    salary: float = Field(gt=0)
    currency: str = Field(default="USD", max_length=10)
    start_date: date
    end_date: Optional[date] = None
    benefits: str = Field(min_length=1)
    reporting_manager: str = Field(min_length=1, max_length=255)
    department: str = Field(min_length=1, max_length=255)
    location: str = Field(min_length=1, max_length=255)
    additional_terms: Optional[str] = None


class OfferLetterUpdate(BaseModel):
    salary: Optional[float] = Field(default=None, gt=0)
    currency: Optional[str] = Field(default=None, max_length=10)
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    benefits: Optional[str] = None
    reporting_manager: Optional[str] = Field(default=None, max_length=255)
    department: Optional[str] = Field(default=None, max_length=255)
    location: Optional[str] = Field(default=None, max_length=255)
    additional_terms: Optional[str] = None


class OfferLetterApproval(BaseModel):
    approved: bool
    rejection_reason: Optional[str] = None


class OfferLetterResponse(BaseModel):
    id: int
    interview_id: int
    candidate_id: int
    job_id: int
    created_by: int
    approved_by: Optional[int] = None
    status: OfferLetterStatus
    salary: float
    currency: str
    start_date: date
    end_date: Optional[date] = None
    benefits: str
    reporting_manager: str
    department: str
    location: str
    additional_terms: Optional[str] = None
    rejection_reason: Optional[str] = None
    sent_at: Optional[datetime] = None
    approved_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    # Enriched fields from relationships
    candidate_name: Optional[str] = None
    candidate_email: Optional[str] = None
    job_title: Optional[str] = None

    model_config = {"from_attributes": True}


class OfferLetterListResponse(BaseModel):
    items: List[OfferLetterResponse]
    total: int
    page: int
    page_size: int

File: frontend/src/types/offerLetter.ts
export type OfferLetterStatus =
  | 'draft'
  | 'pending_approval'
  | 'approved'
  | 'rejected'
  | 'sent';

export interface OfferLetter {
  id: number;
  interview_id: number;
  candidate_id: number;
  job_id: number;
  created_by: number;
  approved_by?: number | null;
  status: OfferLetterStatus;
  salary: number;
  currency: string;
  start_date: string;
  end_date?: string | null;
  benefits: string;
  reporting_manager: string;
  department: string;
  location: string;
  additional_terms?: string | null;
  rejection_reason?: string | null;
  sent_at?: string | null;
  approved_at?: string | null;
  created_at: string;
  updated_at: string;
  candidate_name?: string | null;
  candidate_email?: string | null;
  job_title?: string | null;
}

export interface OfferLetterCreate {
  interview_id: number;
  salary: number;
  currency: string;
  start_date: string;
  end_date?: string;
  benefits: string;
  reporting_manager: string;
  department: string;
  location: string;
  additional_terms?: string;
}

export interface OfferLetterUpdate {
  salary?: number;
  currency?: string;
  start_date?: string;
  end_date?: string;
  benefits?: string;
  reporting_manager?: string;
  department?: string;
  location?: string;
  additional_terms?: string;
}

export interface OfferLetterApproval {
  approved: boolean;
  rejection_reason?: string;
}

export interface OfferLetterListResponse {
  items: OfferLetter[];
  total: number;
  page: number;
  page_size: number;
}
